### PR TITLE
PEAR-2030: Fix for Cohort Builder, number ranges, difficult to type in a number

### PR DESCRIPTION
**Pull request by paribartandhakal**
## Description
Includes refactor for `NumericRangeFacet.tsx`

_**PLEASE VERIFY ALL THE NUMERIC RANGE FACETS IF POSSIBLE**_ 🙏 

## Checklist
- [X] Added proper unit tests
- [N/A] Left proper TODO messages for any remaining tasks
- [N/A] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

**Comment by wteouchicago** [comment]
1. Minor The word "range" is a bit weird for these, esp now that we have added "Min" and "Max". Can we drop it? I think "Min: 0" / "Max: 999999" should convey whatever we need. cc @ebusygina 

Screenshot 2024-08-05 at 1 00 34 PM

__
Looks fine otherwise. 👍 Evgeniia can test more once we are on qa.

**Comment by paribartandhakal** [comment]
> 1. Minor The word "range" is a bit weird for these, esp now that we have added "Min" and "Max". Can we drop it? I think "Min: 0" / "Max: 999999" should convey whatever we need. cc @ebusygina
> 
> Screenshot 2024-08-05 at 1 00 34 PM
> __ Looks fine otherwise. 👍 Evgeniia can test more once we are on qa.

Just updated it.

**Comment by wteouchicago** [comment]
looks good. 👍

File: packages/portal-proto/src/features/facets/NumericRangeFacet/FromTo.tsx [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+const WarningOrError: React.FC<WarningOrErrorProps> = ({
+  hasErrors,
+  isWarning,
+  lowerUnitRange,
+  upperUnitRange,
+}: WarningOrErrorProps) => (
+  <div className="flex flex-col gap-1 my-1">
+    {hasErrors ? (
+      <span className="text-[#AD2B4A]">
```
Comment (craigrbarnes) [inline packages/portal-proto/src/features/facets/NumericRangeFacet/FromTo.tsx]: We define the tailwind color "utility-error" which would be better to use.

**Comment by craigrbarnes** [comment]
Also, it is possible to vertically align the icon and the message 
The icon seems too small to me; I find it hard to see the **!**  (JC might think so too).

File: packages/portal-proto/src/features/facets/NumericRangeFacet/FromTo.tsx (hunk 1 above)
Comment (paribartandhakal) [reply packages/portal-proto/src/features/facets/NumericRangeFacet/FromTo.tsx]: I updated this.

**Comment by paribartandhakal** [comment]
> Also, it is possible to vertically align the icon and the message The icon seems too small to me; I find it hard to see the **!** (JC might think so too).

I have made the changes but waiting for @cemile-ctds' verification.

**Review by craigrbarnes — APPROVED** [review]
The refactoring is quite extensive, but overall, it's fine.

**Pull request merged**